File: kiki616/ABC/B/313.py
N,M=map(int,input().split())
l=list(range(1,N+1))
for i in []*M:
    A,B=map(int,input().split())
    if B in l:
        l.remove(B)
print(l[0] if len(l)==1 else -1)

File: OCH_Tao/ABC/A/313.js
const input = (require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n"));
const N = parseInt(input[0], 10);
const P = input[1].split(" ").map(x => parseInt(x, 10));
if (N === 1) {
  console.log(0);
} else {
  if (P[0] > Math.max(...P.slice(1))) {
    console.log(0);
  } else {
    console.log(Math.max(...P.slice(1)) - P[0] + 1);
  }
}